### ISSUE TITLE
Opal hotreload throws an exception after upgrading to opal 1.0, when reloading a file with unicode character © 

### ISSUE THREAD

**Issue by mpantel** [hyperstack-org/hyperstack#366]
I am on hyperspec-refactor branch and just upgraded a hyperstack application to opal 1.0. I noticed for specific cases hotreload crashes on the browser failing to update the running code. 

The message i get goes like this:
```
Uncaught 
Object { name: "SyntaxError", message: "malformed UTF-8 character", backtrace: (2) […], stack: "(eval):in `end'\n" }
hotloader.self-87cd71ddd8c01b3dd01ee6bed8eeb3e7ff2d08665f751d8d294b6db63d20cffe.js:223:22
Reloading /root/app/hyperstack/components/conference/program.rb (asset_path: components/conference/program.rb)
io.self-50791bd9d612afe52006167d6508c858ad0df7555da96fa23a0cc63ef719ccce.js:156:143

io.self-50791bd9d612afe52006167d6508c858ad0df7555da96fa23a0cc63ef719ccce.js:156:143
Empty string passed to getElementById(). page.self-75242524e5123b5194addca80b0a6b44520df6123a2812ee64d523c3d7635e98.js:37:19
Reloading /root/app/hyperstack/components/conference/page.rb (asset_path: components/conference/page.rb)
io.self-50791bd9d612afe52006167d6508c858ad0df7555da96fa23a0cc63ef719ccce.js:156:143
Uncaught 
Object { name: "SyntaxError", message: "malformed UTF-8 character", backtrace: (2) […], stack: "(eval):in `end'\n" }
```
The exception is thrown around here:

https://github.com/hyperstack-org/hyperstack/blob/c0f18b016124ba66a8c14bb65d27cfd561039f75/ruby/hyperstack-config/lib/hyperstack/hotloader.rb#L93

After some debugging i found that in the reloading Ruby file a unicode character "©" existed in a string to be rendered. The interesting part is that it causes an error even when the line is commented out. 
Removal of that character fixes the problem for hotreload. 

Normal loading works as expected. 

Can anyone help with the actual cause of the problem?

**Comment by mpantel** [comment]
Maybe it is relevand with this https://github.com/opal/opal/pull/2117

At first i've tried to patch hotreload in config/initializers/hyperstack.rb with the following code, but didn't work:
```
module Hyperstack
  class Hotloader
    def reload(e)
      reload_request = JSON.parse(`e.data`)
      if reload_request[:type] == "ruby"
        puts "Reloading #{reload_request[:filename]} (asset_path: #{reload_request[:asset_path]})"
        begin
          #Hyperstack::Context.reset! false
          file_name = reload_request[:asset_path].gsub(/.+hyperstack\//, '')  # this gsub we need sometimes????
          Hotloader.remove(file_name)
          $eval_proc.call file_name, reload_request[:source_code].force_encoding('UTF-8') #<---------
        rescue
          notify_error(reload_request)
        end
        if @reload_post_callback
          @reload_post_callback.call
        else
          puts "no reloading callback to call"
        end
      end
      if reload_request[:type] == "css"
        @css_reloader.reload(reload_request, `document`)
      end
    end
  end
end
```
I Supposed it will be fixed in more recent versions of opal ~>1.1
opal-jquery and opal-sprockets keep opal ~> 1.0

**Comment by mpantel** [comment]
Solved by https://github.com/hyperstack-org/hyperstack/issues/367
